distributed/scheduler.py

```diff
@@ -2126,6 +2126,18 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
                 ws = min(worker_pool.values(), key=operator.attrgetter("occupancy"))
+                if ws.occupancy == 0:  # special case to use round-robin
+                    wp_vals = worker_pool.values()
+                    start = self._n_tasks % n_workers
```

Comment:
Would annotate these. The result when Cythonized is a fast C-style `for`-loop

```suggestion
                    start: Py_ssize_t = self._n_tasks % n_workers
                    i: Py_ssize_t
```